src/render.js

```diff
@@ -42,104 +42,74 @@ const renderToSketch = (
   return renderLayers([layer], container);
 };
 
-// Crawl tree data to find all <Page> components
-const findPageData = (
-  current,
-  depth = 0,
-  accumulated = []
-): Array<{ type: string, children: Object, name?: string }> => {
-  const children = current.children || [];
-  for (let i = 0, len = children.length; i < len; i += 1) {
-    const node = children[i];
-
-    if (node.type === 'page') {
-      accumulated.push({
-        type: 'page',
-        name: node.props.name,
-        children: node.children,
-      });
-    }
-
-    findPageData(children[i], depth + 1);
-  }
-  return accumulated;
-};
-
-const buildPages = (
+const renderContainer = (
   tree: TreeNode,
-  container: ?SketchLayer
+  container: SketchLayer
 ): ?SketchLayer | Array<?SketchLayer> => {
-  const pageData = findPageData(tree);
-  const symbolPage = getSymbolsPage();
+  resetLayer(container);
 
-  if (pageData.length === 0) {
-    const _container = container || getDocument(context).currentPage();
-    const page =
-      !symbolPage || _container !== symbolPage
-        ? _container
-        : getDocument(context).addBlankPage();
+  if (tree.type === 'page') {
+    // assume if name is set on this nested page, the intent is to overwrite
+    // the name of the page it is getting rendered into
+    if (tree.props.name) {
+      container.setName(tree.props.name);
+    }
 
-    return renderToSketch(tree, page);
+    return tree.children.map(child => renderToSketch(child, container));
   }
+  return renderToSketch(tree, container);
+};
 
-  // Keep track of created pages
-  // Starts at `1` by default, because there is always one default page per document
-  let pageTotal = symbolPage ? 2 : 1;
-  // Keep track of existing and created pages to pass back to function caller
-  const pages = [];
-
-  pageData.forEach((data) => {
-    // Get Current Page
-    let page = getDocument(context).currentPage();
-
-    if (pageTotal > 1) {
-      // Create new page
-      page = getDocument(context).addBlankPage();
-    } else {
-      pageTotal += 1;
-    }
+export const renderDocument = (tree: TreeNode): Array<?SketchLayer> => {
+  resetDocument();
 
-    if (data.name) {
-      // Name new page
-      page.setName(data.name);
-    }
+  const doc = getDocument(context);
+  const initialPage = doc.currentPage();
+  const shouldRenderInitialPage = isSymbolsPage(initialPage);
 
-    if (data.children && data.children.length > 0) {
-      // Clear out page layers to prepare for re-render
-      resetLayer(page);
-      data.children.forEach((child) => {
-        renderToSketch(child, page);
-      });
-    }
+  return tree.children.map((child, i) => {
+    const page =
```

Comment:
shall we add a check to make sure the child is a Page?

Reply:
We can... would you want to throw if it isn't?

Reply:
Yeah that'd be nice